test/boostTest.cpp

```diff
@@ -86,14 +86,30 @@ int registerTests(
 			fs::directory_iterator(fullpath),
 			fs::directory_iterator()
 		))
-			if (fs::is_directory(entry.path()) || TestCase::isTestFilename(entry.path().filename()))
+		{
+			bool addTest = true;
+			bool insideSemanticTests = false;
+			fs::path parent_path;
+			for (auto const& element: entry.path())
+			{
+				parent_path = parent_path / element;
+				if (boost::ends_with(canonical(parent_path).generic_string(), "test/libsolidity/semanticTests"))
```

Comment:
Are you sure that this works properly? I think path will not end with "test/libsolidity/semanticTests" as you already added `element` to `parent_path`